src/lib/index/wb_cache.cpp

```diff
@@ -102,16 +101,31 @@ void IndexWBCache::realloc_buf(const IndexBufferPtr& buf) {
 }
 
 void IndexWBCache::write_buf(const BtreeNodePtr& node, const IndexBufferPtr& buf, CPContext* cp_ctx) {
+    // TODO upsert always returns false even if it succeeds.
     m_cache.upsert(node);
     r_cast< IndexCPContext* >(cp_ctx)->add_to_dirty_list(buf);
     resource_mgr().inc_dirty_buf_size(m_node_size);
 }
 
-IndexBufferPtr IndexWBCache::copy_buffer(const IndexBufferPtr& cur_buf) const {
-    auto new_buf = std::make_shared< IndexBuffer >(cur_buf->m_blkid, m_node_size, m_vdev->align_size());
-    std::memcpy(new_buf->raw_buffer(), cur_buf->raw_buffer(), m_node_size);
-    LOGTRACEMOD(wbcache, "new_buf {} cur_buf {} cur_buf_blkid {}", static_cast< void* >(new_buf.get()),
-                static_cast< void* >(cur_buf.get()), cur_buf->m_blkid.to_integer());
+IndexBufferPtr IndexWBCache::copy_buffer(const IndexBufferPtr& cur_buf, const CPContext* cp_ctx) const {
+    IndexBufferPtr new_buf = nullptr;
+    bool copied = false;
+
+    // When we copy the buffer we check if the node buffer is clean or not. If its clean
+    // we could reuse it otherwise create a copy.
+    if (cur_buf->is_clean()) {
+        // Refer to the same node buffer.
+        new_buf = std::make_shared< IndexBuffer >(cur_buf->m_node_buf, cur_buf->m_blkid);
+    } else {
+        // If its not clean, we do deep copy.
+        new_buf = std::make_shared< IndexBuffer >(cur_buf->m_blkid, m_node_size, m_vdev->align_size());
+        std::memcpy(new_buf->raw_buffer(), cur_buf->raw_buffer(), m_node_size);
+        copied = true;
+    }
+
+    LOGTRACEMOD(wbcache, "cp {} new_buf {} cur_buf {} cur_buf_blkid {} copied {}", cp_ctx->id(),
```

Comment:
I doubt whether this is still needed, if not, we are remove a few unused parameters.

Reply:
only for debugging purpose, to check a buffer is copied from another or not. We need a better way to trace than from logs.